OrbitGl/TimeGraph.cpp

```diff
@@ -848,6 +936,43 @@ std::shared_ptr<GpuTrack> TimeGraph::GetOrCreateGpuTrack(
   return track;
 }
 
+//-----------------------------------------------------------------------------
+static void SetTrackNameFromRemoteMemory(std::shared_ptr<Track> track,
+                                         uint64_t string_address,
+                                         ThreadPool* thread_pool) {
+  thread_pool->Schedule([track, string_address]() {
+    int32_t pid = Capture::GTargetProcess->GetID();
+    const auto& process_manager = GOrbitApp->GetProcessManager();
+    auto error_or_string =
+        process_manager->LoadNullTerminatedString(pid, string_address);
+
+    if (error_or_string.has_value()) {
+      GOrbitApp->GetMainThreadExecutor()->Schedule([track, error_or_string]() {
+        track->SetName(error_or_string.value());
+        track->SetLabel(error_or_string.value());
+      });
+    } else {
+      ERROR("Setting track name from remote memory: %s",
+            error_or_string.error().message());
+    }
+  });
+}
+
+//-----------------------------------------------------------------------------
+std::shared_ptr<GraphTrack> TimeGraph::GetOrCreateGraphTrack(
```

Comment:
Does the ownership of the returned `GraphTrack` need to be shared with the caller or can you return a `GraphTrack*`?

Reply:
Well it doesn't need to be shared now no, removing the std::shared_ptr.